src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+(() => {
+  const body = document.querySelector('body');
+
+  let left = false;
+  let right = false;
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+  });
+
+  const notification = (parentNode, classes, text) => {
+    const node = document.createElement('div');
+
+    node.classList.add(...classes);
+    node.textContent = text;
+    node.dataset.qa = 'notification';
+    parentNode.append(node);
+  };
+
+  const firstPromise = new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+      left = true;
+    });
+
+    setTimeout(() => {
+      if (!left) {
+        reject(new Error('First promise was rejected'));
+      }
+    }, 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    document.addEventListener('mousedown', (e) => {
+      if (e.button === 0) {
+        left = true;
+        resolve('Second promise was resolved');
+      }
+
+      if (e.button === 2) {
+        right = true;
+
+        resolve('Second promise was resolved');
+      }
+    });
+  });
```

Comment:
You're resolving the secondPromise twice. Once the promise is resolved, it should not be resolved again. Instead, you should check if both buttons have been clicked before resolving the promise.